cookiecutter/generate.py

```diff
@@ -100,7 +100,13 @@ def generate_context(context_file='cookiecutter.json', default_context=None,
 
     # Add the Python object to the context dictionary
     file_name = os.path.split(context_file)[1]
-    file_stem = file_name.split('.')[0]
+    # Take care of file name starting with a '.'
```

Comment:
This code is kind of complex. Some suggestions:
* Maybe add a comment to explain what it's doing and explain what kind of situations it's intended to handle?
* Consider moving this code to a separate function and adding a few tests for it

Reply:
How about:

```python
# Handle the case for hidden files where a file_name starts with a dot ('.')
file_name_splits = file_name.split('.')
file_stem = file_name_splits[1] if file_name.startswith('.') else file_name_splits[0]
```